src/notification-banner/notification-banner.tsx

```diff
@@ -40,36 +47,94 @@ export const NBComponent = ({
     // =============================================================================
     // EVENT HANDLERS
     // =============================================================================
-    const handleDismiss = () => {
+    const handleDismiss = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
         setVisible(false);
 
         if (dismissible && onDismiss) onDismiss();
     };
 
+    const handleBannerClick = (event: React.MouseEvent) => {
+        if (!onClick) return;
+        event.stopPropagation();
+        onClick();
+    };
+
+    const handleActionButtonOnClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
+        if (!actionButton.onClick) {
+            handleBannerClick(event);
+            return;
+        }
+        actionButton.onClick(event);
+    };
+
     // =============================================================================
     // RENDER FUNCTIONS
     // =============================================================================
     if (!isVisible) return null;
 
+    const renderDismissButton = () => (
+        <StyledIconButton
+            onClick={handleDismiss}
+            id={formatId("dismiss-button", id)}
+            data-testid={formatId("dismiss-button", testId)}
+            focusHighlight={false}
+        >
+            <StyledIcon />
+        </StyledIconButton>
+    );
+
+    const renderActionButton = () => (
+        <ActionButton
+            id={formatId("action-button", id)}
+            data-testid={formatId("action-button", testId)}
+            onClick={handleActionButtonOnClick}
+            {...actionButton}
```

Comment:
the order should be swapped here so that the DS logic takes precedence

```suggestion
            {...actionButton}
            onClick={handleActionButtonOnClick}
```

Reply:
after some testing, I think the click handlers can be simplified

```
// actionButton
if (!actionButton.onClick)
  return // do nothing, let it bubble to the banner
event.stopPropagation()
actionButton.onClick(event);

// accessible banner button
<AccessibleBannerButton /> // render without an onClick and the event bubbles to the banner as well

// handleBannerClick - not required, let the component accept `onClick` directly aka the native behaviour
```